src/main/java/com/cloudbees/jenkins/plugins/mtslavescloud/MansionRetentionStrategy.java

```diff
@@ -114,4 +156,55 @@ protected long getIdleMaxTime() {
 
 
     private static Logger LOGGER  = Logger.getLogger(MansionRetentionStrategy.class.getName());
+    
+    private static final QueueLock queueLock = newQueueLock();
+    
+    private static QueueLock newQueueLock() {
+        try {
+            return new ReflectionPost592QueueLock();
+        } catch (NoSuchMethodException e) {
+            return new Pre592QueueLock();
+        }
+    }
+
+    // TODO replace all this with Queue.withLock once Jenkins 1.592+
```

Comment:
Well, I think you could just upgrade the POM to 1.609 and avoid all this. DEV@cloud should be on 1.609.1 by Monday.